### DISCUSSION TITLE
Subsetting issue on Windows

### DISCUSSION THREAD

**Discussion by imcslatte** [blaylockbk/Herbie#84] [General]
I've installed Herbie on a Windows 10 machine running python 3.10. My objective is to generate regional subsets of surface met variables (ugrd,vgrd,tmp at 2m,sfcpress , etc) for use as forcing for a regional ocean model.  The first step is to subset the data by variable. 

I stared with a simple example from the tutorial

```python
from herbie.archive import Herbie
H = Herbie('2022-7-12 00:00',product='sfc',source='aws')
```

```
✅ Found ┊ model=hrrr ┊ product=sfc ┊ 2022-Jul-12 00:00 UTC F00 ┊ GRIB2 @ aws ┊ IDX @ `aws`
```

```python
searchstring=':TMP:2 m'
H.read_idx(searchstring)
```
<html>
<body>

  | grib_message | start_byte | end_byte | range | reference_time | valid_time | variable | level | forecast_time | search_this
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
71 | 36433970 | 37698543 | 36433970-37698543 | 2022-07-12 | 2022-07-12 | TMP | 2 m above ground | anl | :TMP:2 m above ground:anl

</body>
</html>

```python
H.download(verbose=True)
```
```
✅ Success! Downloaded HRRR from local               
	src: C:\Users\Eli Hunter\data\hrrr\20220712\hrrr.t00z.wrfsfcf00.grib2
	dst: C:\Users\Eli Hunter\data\hrrr\20220712\hrrr.t00z.wrfsfcf00.grib2
WindowsPath('C:/Users/Eli Hunter/data/hrrr/20220712/hrrr.t00z.wrfsfcf00.grib2')
```

Downloading the entire file works fine. It appears in my home directory as expected. 

However, when I try this:
```python
H.download(searchstring,verbose=True)
```
```
📇 Download subset: ▌▌Herbie HRRR model sfc product initialized 2022-Jul-12 00:00 UTC F00 ┊ source=local                                                            
 cURL from [file://C:\Users\Eli](file:///C:/Users/Eli) Hunter\data\hrrr\20220712\hrrr.t00z.wrfsfcf00.grib2
  71  :TMP:2 m above ground:anl
💾 Saved the subset to C:\Users\Eli Hunter\data\hrrr\20220712\subset_b7103ca278a75cad8f7d065acda0c2e80da0b7dc__hrrr.t00z.wrfsfcf00.grib2
WindowsPath('C:/Users/Eli Hunter/data/hrrr/20220712/subset_b7103ca278a75cad8f7d065acda0c2e80da0b7dc__hrrr.t00z.wrfsfcf00.grib2')
```

The output suggests a subset file was downloaded and is now in my home directory. However, the file subset_b7103ca278a75cad8f7d065acda0c2e80da0b7dc__hrrr.t00z.wrfsfcf00.grib2 does not exist. And it was instantaneous. There did not seem to be any delay associated with a file download.  

Is there a configurations step I am missing? 

Thanks.,
Eli

**Comment by blaylockbk** [comment]
I just tried this on Linux and didn't have an issue. I have both these files in my `~/data/hrrr/20220712` directory.
```
hrrr.t00z.wrfsfcf00.grib2 
subset_b7103ca278a75cad8f7d065acda0c2e80da0b7dc__hrrr.t00z.wrfsfcf00.grib2
```

I'll have to try it on my PC to see if I can reproduce your issue.

**Comment by blaylockbk** [comment]
I wasn't able to reproduce your issue on my Windows

<img width="827" alt="image" src="https://user-images.githubusercontent.com/6249613/181675589-a9586f9d-a5c1-463e-b359-3aec60e187de.png">
<img width="818" alt="image" src="https://user-images.githubusercontent.com/6249613/181675622-cc103f86-4bf1-4aa0-8e5a-9913db6226cb.png">

And both files appear in the file explorer
<img width="707" alt="image" src="https://user-images.githubusercontent.com/6249613/181675659-774e18e6-cec3-4d1b-8bfc-0a5e75d489f5.png">

**Comment by blaylockbk** [comment]
What do you get when you execute
```python
H.xarray(searchstring)
```
you should get the 2-m temperature data read into an xarray DataSet

<img width="582" alt="image" src="https://user-images.githubusercontent.com/6249613/181676034-57993b73-d2df-49a8-a3f7-deacdf24ea5a.png">
